### PR TITLE
Cellular: update handbook to 5.11

### PR DESCRIPTION
**Pull request by jarvte**
Updated cellular parts to match current implementation after refactoring.

@AriParkkila please review.

### CODE REVIEW

**Comment by jarvte** [comment]
@AnotherButler please review

**Comment by jarvte** [comment]
@ARMmbed/mbed-docs please review as Amanda is out of office.

**Comment by iriark01** [comment]
This will probably have to wait for Amanda. We have a release and an OOB campaign this week for other products.

**Comment by AnotherButler** [comment]
Is there a code dependency on this PR?

File: docs/api/networkinterfaces/CellularInterface.md [hunk 1]
```diff
@@ -4,13 +4,13 @@
 
 The CellularBase provides a C++ API for connecting to the internet over a Cellular device.
 
-Arm Mbed OS provides a [reference implementation of CellularBase](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/_easy_cellular_connection_8h_source.html), which has more information.
+Arm Mbed OS provides a [reference implementation of CellularBase](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/_cellular_context_8h_source.html), which has more information.
```
Comment (AnotherButler) [inline docs/api/networkinterfaces/CellularInterface.md]: Query: Why do we link to the same Doxy page three times on the same page? Could we get rid of two?

File: docs/reference/technology/connectivity/cellular.md [hunk 2]
```diff
@@ -119,7 +136,7 @@ Consider the following points when selecting PPP or AT mode:
 
 ### Optimize for power consumption
 
-The `CellularPower` class has methods to optimize power saving. The `set_powerl_level()` offers flexibility to control the reception and transmission power levels. In addition, 3GPP has specified advanced power optimizations that are useful for celluar IoT devices: Power Save Mode (PSM) and extended Discontinuous Reception (eDRX).
+The `CellularPower` class has methods to optimize power saving. The `set_power_level()` offers flexibility to control the reception and transmission power levels. In addition, 3GPP has specified advanced power optimizations that are useful for celluar IoT devices: Power Save Mode (PSM) and extended Discontinuous Reception (eDRX).
```
Comment (AnotherButler) [inline docs/reference/technology/connectivity/cellular.md]: Good catch 👍

File: docs/reference/technology/connectivity/cellular.md [hunk 3]
```diff
@@ -48,33 +48,51 @@ If you use an Mbed OS target and a separate cellular hosted module via a serial
...
 
 ### Cellular APIs
 
-As an application developer, you should use and refer only to classes located under API folder. All the other classes have implementation details which are expected to change frequently.
+As an application developer, you should use and refer only to classes located under API folder. All the other classes have implementation details that are expected to change frequently.
 
 Cellular APIs are structured based on main functionalities:
 
-- `CellularNetwork` for cellular network features, such as preferred operator and APN.
+- `CellularContext` is the main interface for the application. You can use it to connect to the operator's APN.
```
Comment (AnotherButler) [inline docs/reference/technology/connectivity/cellular.md]: Query: What does APN stand for?

Comment (jarvte) [reply docs/reference/technology/connectivity/cellular.md]: Access Point Name: https://en.wikipedia.org/wiki/Access_Point_Name

**Comment by jarvte** [comment]
> Is there a code dependency on this PR?
These changes are related to this pr which is merged: https://github.com/ARMmbed/mbed-os/pull/8579

File: docs/api/networkinterfaces/CellularInterface.md (hunk 1 above)
Comment (jarvte) [reply docs/api/networkinterfaces/CellularInterface.md]: Good point, removed two links.

**Pull request merged**